docker.go

```diff
@@ -46,9 +46,17 @@ type Parameters struct {
 
 // Volume is the docker volume details associated to the container
 type Volume struct {
-	ContainerPath string `json:"containerPath,omitempty"`
-	HostPath      string `json:"hostPath,omitempty"`
-	Mode          string `json:"mode,omitempty"`
+	ContainerPath  string          `json:"containerPath,omitempty"`
+	HostPath       string          `json:"hostPath,omitempty"`
+	ExternalVolume *ExternalVolume `json:"external,omitempty"`
+	Mode           string          `json:"mode,omitempty"`
+}
+
+// An external volume definition
+type ExternalVolume struct {
+	Name     string             `json:"name,omitempty"`
+	Provider string             `json:"provider,omitempty"`
```

Comment:
Looking at the json schema in http://mesosphere.github.io/marathon/docs/generated/api.html#v2_apps__app_id__put it seems like Name and Provider are, at least from an API definition point of view, not required to have a value. 
So I'm inclined to leave these the way they are. In my use-case they definitely always have a value, but there might be some usage out there that doesn't require this.